sklearn/model_selection/tests/test_search.py

```diff
@@ -173,6 +173,28 @@ def test_grid_search():
     assert_raises(ValueError, grid_search.fit, X, y)
 
 
+def test_grid_search_with_fit_params():
+    X = np.arange(100).reshape(10, 10)
+    y = np.array([0] * 5 + [1] * 5)
+    clf = CheckingClassifier(expected_fit_params=['spam', 'eggs'])
+    grid_search = GridSearchCV(clf, {'foo_param': [1, 2, 3]})
+
+    # The CheckingClassifer generates an assertion error if
+    # a parameter is missing or has length != len(X).
```

Comment:
Instead of putting a comment to explain what CheckingClassifier should raise in that case, please add an assertion with `assert_raises`.